diff -ruN xmrig-proxy/CMakeLists.txt xmrig-proxy_new/CMakeLists.txt
--- xmrig-proxy/CMakeLists.txt	2018-03-25 20:43:48.979370716 +0800
+++ xmrig-proxy_new/CMakeLists.txt	2018-03-25 20:53:41.299662241 +0800
@@ -18,7 +18,6 @@
     src/core/ConfigLoader_static.h
     src/core/ConfigWatcher.h
     src/core/Controller.h
-    src/donate.h
     src/interfaces/IClientListener.h
     src/interfaces/IConsoleListener.h
     src/interfaces/IControllerListener.h
@@ -37,7 +36,6 @@
     src/net/Client.h
     src/net/Id.h
     src/net/Job.h
-    src/net/strategies/DonateStrategy.h
     src/net/strategies/FailoverStrategy.h
     src/net/strategies/SinglePoolStrategy.h
     src/net/Url.h
@@ -93,7 +91,6 @@
     src/log/ShareLog.cpp
     src/net/Client.cpp
     src/net/Job.cpp
-    src/net/strategies/DonateStrategy.cpp
     src/net/strategies/FailoverStrategy.cpp
     src/net/strategies/SinglePoolStrategy.cpp
     src/net/SubmitResult.cpp
diff -ruN xmrig-proxy/README.md xmrig-proxy_new/README.md
--- xmrig-proxy/README.md	2018-03-25 20:43:48.979370716 +0800
+++ xmrig-proxy_new/README.md	2018-03-25 20:54:30.470512793 +0800
@@ -46,7 +46,6 @@
       --no-color           disable colored output
       --no-workers         disable per worker statistics
       --variant            algorithm PoW variant
-      --donate-level=N     donate level, default 2%
   -B, --background         run the miner in the background
   -c, --config=FILE        load a JSON-format configuration file
       --no-watch           disable configuration file watching
@@ -62,12 +61,6 @@
   -V, --version            output version information and exit
 ```
 
-## Donations
-
-Default donation fee is 2% can be reduced to 1% or disabled via `donate-level` option. Donation fee applies only if you use more than 256 miners.
-
-* XMR: `48edfHu7V9Z84YzzMa6fUueoELZ9ZRXq9VetWzYGzKt52XU5xvqgzYnDK9URnRoJMk1j8nLwEVsaSWJ4fhdUyZijBGUicoD`
-* BTC: `1P7ujsXeX7GxQwHNnJsRMgAdNkFZmNVqJT`
 
 ## Release checksums
 ### SHA-256
diff -ruN xmrig-proxy/src/api/ApiRouter.cpp xmrig-proxy_new/src/api/ApiRouter.cpp
--- xmrig-proxy/src/api/ApiRouter.cpp	2018-03-25 20:43:48.985370942 +0800
+++ xmrig-proxy_new/src/api/ApiRouter.cpp	2018-03-25 20:56:13.046373433 +0800
@@ -212,14 +212,6 @@
     doc.AddMember("mode",         rapidjson::StringRef(m_controller->config()->modeName()), allocator);
     doc.AddMember("ua",           rapidjson::StringRef(Platform::userAgent()), allocator);
     doc.AddMember("uptime",       stats.uptime(), allocator);
-    doc.AddMember("donate_level", m_controller->config()->donateLevel(), allocator);
-
-    if (stats.hashes && stats.donateHashes) {
-        doc.AddMember("donated", normalize((double) stats.donateHashes / stats.hashes * 100.0), allocator);
-    }
-    else {
-        doc.AddMember("donated", 0.0, allocator);
-    }
 }
 
 
@@ -278,7 +270,6 @@
     results.AddMember("avg_time",      stats.avgTime(), allocator);
     results.AddMember("latency",       stats.avgLatency(), allocator);
     results.AddMember("hashes_total",  stats.hashes, allocator);
-    results.AddMember("hashes_donate", stats.donateHashes, allocator);
 
     rapidjson::Value best(rapidjson::kArrayType);
     for (size_t i = 0; i < stats.topDiff.size(); ++i) {
diff -ruN xmrig-proxy/src/config.json xmrig-proxy_new/src/config.json
--- xmrig-proxy/src/config.json	2018-03-25 20:43:48.985370942 +0800
+++ xmrig-proxy_new/src/config.json	2018-03-25 20:56:35.765229625 +0800
@@ -15,7 +15,6 @@
     ],
     "colors": true,
     "custom-diff": 0,
-    "donate-level": 2,
     "log-file": null,
     "mode": "nicehash",
     "pools": [
@@ -33,4 +32,4 @@
     "verbose": false,
     "watch": true,
     "workers": true
-}
\ No newline at end of file
+}
diff -ruN xmrig-proxy/src/core/Config.cpp xmrig-proxy_new/src/core/Config.cpp
--- xmrig-proxy/src/core/Config.cpp	2018-03-25 20:43:48.985370942 +0800
+++ xmrig-proxy_new/src/core/Config.cpp	2018-03-25 21:15:49.452703926 +0800
@@ -28,7 +28,6 @@
 
 #include "core/Config.h"
 #include "core/ConfigLoader.h"
-#include "donate.h"
 #include "log/Log.h"
 #include "net/Url.h"
 #include "proxy/Addr.h"
@@ -79,7 +78,6 @@
     m_userAgent(nullptr),
     m_algorithm(CRYPTONIGHT),
     m_apiPort(0),
-    m_donateLevel(kDonateLevel),
     m_mode(NICEHASH_MODE),
     m_retries(2),
     m_retryPause(1),
@@ -197,7 +195,6 @@
     doc.AddMember("bind",         bind, allocator);
     doc.AddMember("colors",       colors(), allocator);
     doc.AddMember("custom-diff",  diff(), allocator);
-    doc.AddMember("donate-level", donateLevel(), allocator);
     doc.AddMember("log-file",     logFile() ? rapidjson::Value(rapidjson::StringRef(logFile())).Move() : rapidjson::Value(rapidjson::kNullType).Move(), allocator);
     doc.AddMember("mode",         rapidjson::StringRef(modeName()), allocator);
 
diff -ruN xmrig-proxy/src/core/Config.h xmrig-proxy_new/src/core/Config.h
--- xmrig-proxy/src/core/Config.h	2018-03-25 20:43:48.985370942 +0800
+++ xmrig-proxy_new/src/core/Config.h	2018-03-25 21:05:57.520399133 +0800
@@ -101,7 +101,6 @@
     inline const std::vector<Url*> &pools() const  { return m_pools; }
     inline int algorithm() const                   { return m_algorithm; }
     inline int apiPort() const                     { return m_apiPort; }
-    inline int donateLevel() const                 { return m_donateLevel; }
     inline int mode() const                        { return m_mode; }
     inline int retries() const                     { return m_retries; }
     inline int retryPause() const                  { return m_retryPause; }
@@ -137,7 +136,6 @@
     char *m_userAgent;
     int m_algorithm;
     int m_apiPort;
-    int m_donateLevel;
     int m_mode;
     int m_retries;
     int m_retryPause;
diff -ruN xmrig-proxy/src/core/ConfigLoader.cpp xmrig-proxy_new/src/core/ConfigLoader.cpp
--- xmrig-proxy/src/core/ConfigLoader.cpp	2018-03-25 20:43:48.985370942 +0800
+++ xmrig-proxy_new/src/core/ConfigLoader.cpp	2018-03-25 21:04:57.645142845 +0800
@@ -327,15 +327,6 @@
     case 4003: /* --api-no-ipv6 */
         return parseBoolean(config, key, false);
 
-    case 1003: /* --donate-level */
-        if (strncmp(arg, "minemonero.pro", 14) == 0) {
-            config->m_donateLevel = 0;
-        }
-        else {
-            parseArg(config, key, strtol(arg, nullptr, 10));
-        }
-        break;
-
     case 1104: /* --coin */
         config->setCoin(arg);
         break;
@@ -387,14 +378,6 @@
         config->m_retryPause = (int) arg;
         break;
 
-    case 1003: /* --donate-level */
-        if ((int) arg < 0 || arg > 99) {
-            return true;
-        }
-
-        config->m_donateLevel = (int) arg;
-        break;
-
     case 1010: /* --variant */
         config->m_pools.back()->setVariant((int) arg);
         break;
diff -ruN xmrig-proxy/src/core/ConfigLoader_static.h xmrig-proxy_new/src/core/ConfigLoader_static.h
--- xmrig-proxy/src/core/ConfigLoader_static.h	2018-03-25 20:43:48.985370942 +0800
+++ xmrig-proxy_new/src/core/ConfigLoader_static.h	2018-03-25 21:01:16.020791175 +0800
@@ -63,7 +63,6 @@
       --no-color           disable colored output\n\
       --no-workers         disable per worker statistics\n\
       --variant            algorithm PoW variant\n\
-      --donate-level=N     donate level, default 2%%\n\
   -B, --background         run the miner in the background\n\
   -c, --config=FILE        load a JSON-format configuration file\n\
       --no-watch           disable configuration file watching\n\
@@ -101,7 +100,6 @@
     { "config",            1, nullptr, 'c'  },
     { "custom-diff",       1, nullptr, 1102 },
     { "debug",             0, nullptr, 1101 },
-    { "donate-level",      1, nullptr, 1003 },
     { "help",              0, nullptr, 'h'  },
     { "keepalive",         0, nullptr ,'k'  },
     { "log-file",          1, nullptr, 'l'  },
@@ -134,7 +132,6 @@
     { "colors",           0, nullptr, 2000 },
     { "custom-diff",      1, nullptr, 1102 },
     { "debug",            0, nullptr, 1101 },
-    { "donate-level",     1, nullptr, 1003 },
     { "log-file",         1, nullptr, 'l'  },
     { "retries",          1, nullptr, 'r'  },
     { "retry-pause",      1, nullptr, 'R'  },
diff -ruN xmrig-proxy/src/donate.h xmrig-proxy_new/src/donate.h
--- xmrig-proxy/src/donate.h	2018-03-25 20:43:48.986370980 +0800
+++ xmrig-proxy_new/src/donate.h	1970-01-01 08:00:00.000000000 +0800
@@ -1,47 +0,0 @@
-/* XMRig
- * Copyright 2010      Jeff Garzik <jgarzik@pobox.com>
- * Copyright 2012-2014 pooler      <pooler@litecoinpool.org>
- * Copyright 2014      Lucas Jones <https://github.com/lucasjones>
- * Copyright 2014-2016 Wolf9466    <https://github.com/OhGodAPet>
- * Copyright 2016      Jay D Dee   <jayddee246@gmail.com>
- * Copyright 2016-2017 XMRig       <support@xmrig.com>
- *
- *
- *   This program is free software: you can redistribute it and/or modify
- *   it under the terms of the GNU General Public License as published by
- *   the Free Software Foundation, either version 3 of the License, or
- *   (at your option) any later version.
- *
- *   This program is distributed in the hope that it will be useful,
- *   but WITHOUT ANY WARRANTY; without even the implied warranty of
- *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- *   GNU General Public License for more details.
- *
- *   You should have received a copy of the GNU General Public License
- *   along with this program. If not, see <http://www.gnu.org/licenses/>.
- */
-
-#ifndef __DONATE_H__
-#define __DONATE_H__
-
-
-/*
- * Dev donation.
- *
- * Percentage of your hashing power that you want to donate to the developer, can be 0 if you don't want to do that.
- *
- * If you plan on changing this setting to 0 please consider making a one off donation to my wallet:
- * XMR: 48edfHu7V9Z84YzzMa6fUueoELZ9ZRXq9VetWzYGzKt52XU5xvqgzYnDK9URnRoJMk1j8nLwEVsaSWJ4fhdUyZijBGUicoD
- * BTC: 1P7ujsXeX7GxQwHNnJsRMgAdNkFZmNVqJT
- *
- * How it works:
- * First pool connection (up to 256 workers) always without fee.
- * Other connections first randomly switch to dev pool in range from 50 to 150 minutes, to reduce dev pool peak load.
- * Stays on dev pool at least kDonateLevel minutes.
- * Choice next donation time, with overime compensation. In proxy no way to use precise donation time.
- * You can check actual donation via API.
- */
-constexpr const int kDonateLevel = 2;
-
-
-#endif /* __DONATE_H__ */
diff -ruN xmrig-proxy/src/log/ShareLog.cpp xmrig-proxy_new/src/log/ShareLog.cpp
--- xmrig-proxy/src/log/ShareLog.cpp	2018-03-25 20:43:48.987371017 +0800
+++ xmrig-proxy_new/src/log/ShareLog.cpp	2018-03-25 20:57:44.773830289 +0800
@@ -95,10 +95,6 @@
 
 void ShareLog::reject(const AcceptEvent *event)
 {
-    if (event->isDonate()) {
-        return;
-    }
-
     LOG_INFO(isColors() ? "#%03u \x1B[01;31mrejected\x1B[0m (%" PRId64 "/%" PRId64 "+%" PRId64 ") diff \x1B[01;37m%u\x1B[0m ip \x1B[01;37m%s \x1B[31m\"%s\"\x1B[0m \x1B[01;30m(%" PRId64 " ms)"
                         : "#%03u rejected (%" PRId64 "/%" PRId64 "+%" PRId64 ") diff %u ip %s \"%s\" (%" PRId64 " ms)",
              event->mapperId(), m_stats.data().accepted, m_stats.data().rejected, m_stats.data().invalid, event->result.diff, event->ip(), event->error(), event->result.elapsed);
diff -ruN xmrig-proxy/src/net/strategies/DonateStrategy.cpp xmrig-proxy_new/src/net/strategies/DonateStrategy.cpp
--- xmrig-proxy/src/net/strategies/DonateStrategy.cpp	2018-03-25 20:43:48.987371017 +0800
+++ xmrig-proxy_new/src/net/strategies/DonateStrategy.cpp	1970-01-01 08:00:00.000000000 +0800
@@ -1,170 +0,0 @@
-/* XMRig
- * Copyright 2010      Jeff Garzik <jgarzik@pobox.com>
- * Copyright 2012-2014 pooler      <pooler@litecoinpool.org>
- * Copyright 2014      Lucas Jones <https://github.com/lucasjones>
- * Copyright 2014-2016 Wolf9466    <https://github.com/OhGodAPet>
- * Copyright 2016      Jay D Dee   <jayddee246@gmail.com>
- * Copyright 2017-2018 XMR-Stak    <https://github.com/fireice-uk>, <https://github.com/psychocrypt>
- * Copyright 2016-2018 XMRig       <https://github.com/xmrig>, <support@xmrig.com>
- *
- *   This program is free software: you can redistribute it and/or modify
- *   it under the terms of the GNU General Public License as published by
- *   the Free Software Foundation, either version 3 of the License, or
- *   (at your option) any later version.
- *
- *   This program is distributed in the hope that it will be useful,
- *   but WITHOUT ANY WARRANTY; without even the implied warranty of
- *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- *   GNU General Public License for more details.
- *
- *   You should have received a copy of the GNU General Public License
- *   along with this program. If not, see <http://www.gnu.org/licenses/>.
- */
-
-
-#include "core/Config.h"
-#include "core/Controller.h"
-#include "interfaces/IStrategyListener.h"
-#include "net/Client.h"
-#include "net/strategies/DonateStrategy.h"
-#include "Platform.h"
-#include "proxy/StatsData.h"
-
-
-extern "C"
-{
-#include "crypto/c_keccak.h"
-}
-
-
-static inline int random(int min, int max){
-   return min + rand() / (RAND_MAX / (max - min + 1) + 1);
-}
-
-
-DonateStrategy::DonateStrategy(size_t id, xmrig::Controller *controller, IStrategyListener *listener) :
-    m_active(false),
-    m_suspended(false),
-    m_listener(listener),
-    m_id(id),
-    m_donateTicks(0),
-    m_target(0),
-    m_ticks(0),
-    m_controller(controller)
-{
-    uint8_t hash[200];
-    char userId[65] = { 0 };
-    const char *user = controller->config()->pools().front()->user();
-
-    keccak(reinterpret_cast<const uint8_t *>(user), static_cast<int>(strlen(user)), hash, sizeof(hash));
-    Job::toHex(hash, 32, userId);
-
-    Url *url = new Url("proxy.fee.xmrig.com", controller->config()->algorithm() == xmrig::Config::CRYPTONIGHT_LITE ? 7777 : 4444, userId, nullptr);
-
-    m_client = new Client(-1, Platform::userAgent(), this);
-    m_client->setUrl(url);
-    m_client->setRetryPause(1000);
-
-    delete url;
-
-    m_target = random(3000, 9000);
-}
-
-
-DonateStrategy::~DonateStrategy()
-{
-    m_client->deleteLater();
-}
-
-
-bool DonateStrategy::reschedule()
-{
-    const uint64_t level = m_controller->config()->donateLevel() * 60;
-    if (m_donateTicks < level) {
-        return false;
-    }
-
-    m_target = m_ticks + (6000 * ((double) m_donateTicks / level));
-    m_active = false;
-
-    stop();
-    return true;
-}
-
-
-int64_t DonateStrategy::submit(const JobResult &result)
-{
-    return m_client->submit(result);
-}
-
-
-void DonateStrategy::connect()
-{
-    m_suspended = false;
-}
-
-
-void DonateStrategy::stop()
-{
-    m_suspended   = true;
-    m_donateTicks = 0;
-    m_client->disconnect();
-}
-
-
-void DonateStrategy::tick(uint64_t now)
-{
-    m_client->tick(now);
-
-    if (m_suspended) {
-        return;
-    }
-
-    m_ticks++;
-
-    if (m_ticks == m_target) {
-        if (m_id == 0 && m_controller->statsData().upstreams.active == 1) {
-            m_target += 600;
-            return;
-        }
-
-        m_client->connect();
-    }
-
-    if (isActive()) {
-        m_donateTicks++;
-    }
-}
-
-
-void DonateStrategy::onClose(Client *client, int failures)
-{
-    if (!isActive()) {
-        return;
-    }
-
-    m_active = false;
-    m_listener->onPause(this);
-}
-
-
-void DonateStrategy::onJobReceived(Client *client, const Job &job)
-{
-    if (!isActive()) {
-        m_active = true;
-        m_listener->onActive(this, client);
-    }
-
-    m_listener->onJob(this, client, job);
-}
-
-
-void DonateStrategy::onLoginSuccess(Client *client)
-{
-}
-
-
-void DonateStrategy::onResultAccepted(Client *client, const SubmitResult &result, const char *error)
-{
-    m_listener->onResultAccepted(this, client, result, error);
-}
diff -ruN xmrig-proxy/src/net/strategies/DonateStrategy.h xmrig-proxy_new/src/net/strategies/DonateStrategy.h
--- xmrig-proxy/src/net/strategies/DonateStrategy.h	2018-03-25 20:43:48.988371055 +0800
+++ xmrig-proxy_new/src/net/strategies/DonateStrategy.h	1970-01-01 08:00:00.000000000 +0800
@@ -1,76 +0,0 @@
-/* XMRig
- * Copyright 2010      Jeff Garzik <jgarzik@pobox.com>
- * Copyright 2012-2014 pooler      <pooler@litecoinpool.org>
- * Copyright 2014      Lucas Jones <https://github.com/lucasjones>
- * Copyright 2014-2016 Wolf9466    <https://github.com/OhGodAPet>
- * Copyright 2016      Jay D Dee   <jayddee246@gmail.com>
- * Copyright 2017-2018 XMR-Stak    <https://github.com/fireice-uk>, <https://github.com/psychocrypt>
- * Copyright 2016-2018 XMRig       <https://github.com/xmrig>, <support@xmrig.com>
- *
- *   This program is free software: you can redistribute it and/or modify
- *   it under the terms of the GNU General Public License as published by
- *   the Free Software Foundation, either version 3 of the License, or
- *   (at your option) any later version.
- *
- *   This program is distributed in the hope that it will be useful,
- *   but WITHOUT ANY WARRANTY; without even the implied warranty of
- *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- *   GNU General Public License for more details.
- *
- *   You should have received a copy of the GNU General Public License
- *   along with this program. If not, see <http://www.gnu.org/licenses/>.
- */
-
-#ifndef __DONATESTRATEGY_H__
-#define __DONATESTRATEGY_H__
-
-
-#include "interfaces/IClientListener.h"
-#include "interfaces/IStrategy.h"
-
-
-class Client;
-class IStrategyListener;
-class Url;
-
-
-namespace xmrig {
-    class Controller;
-}
-
-
-class DonateStrategy : public IStrategy, public IClientListener
-{
-public:
-    DonateStrategy(size_t id, xmrig::Controller *controller, IStrategyListener *listener);
-    ~DonateStrategy();
-
-    bool reschedule();
-
-    inline bool isActive() const override  { return m_active; }
-    inline void resume() override          {}
-
-    int64_t submit(const JobResult &result) override;
-    void connect() override;
-    void stop() override;
-    void tick(uint64_t now) override;
-
-protected:
-    void onClose(Client *client, int failures) override;
-    void onJobReceived(Client *client, const Job &job) override;
-    void onLoginSuccess(Client *client) override;
-    void onResultAccepted(Client *client, const SubmitResult &result, const char *error) override;
-
-private:
-    bool m_active;
-    bool m_suspended;
-    Client *m_client;
-    IStrategyListener *m_listener;
-    size_t m_id;
-    uint64_t m_donateTicks;
-    uint64_t m_target;
-    uint64_t m_ticks;
-    xmrig::Controller *m_controller;
-};
-
-#endif /* __SINGLEPOOLSTRATEGY_H__ */
diff -ruN xmrig-proxy/src/proxy/events/AcceptEvent.h xmrig-proxy_new/src/proxy/events/AcceptEvent.h
--- xmrig-proxy/src/proxy/events/AcceptEvent.h	2018-03-25 20:43:48.989371093 +0800
+++ xmrig-proxy_new/src/proxy/events/AcceptEvent.h	2018-03-25 21:07:24.115662271 +0800
@@ -35,33 +35,30 @@
 class AcceptEvent : public MinerEvent
 {
 public:
-    static inline bool start(size_t mapperId, Miner *miner, const SubmitResult &result, bool donate, const char *error = nullptr)
+    static inline bool start(size_t mapperId, Miner *miner, const SubmitResult &result, const char *error = nullptr)
     {
-        return exec(new (m_buf) AcceptEvent(mapperId, miner, result, donate, error));
+        return exec(new (m_buf) AcceptEvent(mapperId, miner, result, error));
     }
 
 
     const SubmitResult &result;
 
 
-    inline bool isDonate() const            { return m_donate; }
     inline bool isRejected() const override { return m_error != nullptr; }
     inline const char *error() const        { return m_error; }
     inline size_t mapperId() const          { return m_mapperId; }
 
 
 protected:
-    inline AcceptEvent(size_t mapperId, Miner *miner, const SubmitResult &result, bool donate, const char *error)
+    inline AcceptEvent(size_t mapperId, Miner *miner, const SubmitResult &result, const char *error)
         : MinerEvent(AcceptType, miner),
           result(result),
-          m_donate(donate),
           m_error(error),
           m_mapperId(mapperId)
     {}
 
 
 private:
-    bool m_donate;
     const char *m_error;
     size_t m_mapperId;
 };
diff -ruN xmrig-proxy/src/proxy/splitters/nicehash/NonceMapper.cpp xmrig-proxy_new/src/proxy/splitters/nicehash/NonceMapper.cpp
--- xmrig-proxy/src/proxy/splitters/nicehash/NonceMapper.cpp	2018-03-25 20:43:48.989371093 +0800
+++ xmrig-proxy_new/src/proxy/splitters/nicehash/NonceMapper.cpp	2018-03-25 21:11:23.694690037 +0800
@@ -31,7 +31,6 @@
 #include "core/Controller.h"
 #include "log/Log.h"
 #include "net/Client.h"
-#include "net/strategies/DonateStrategy.h"
 #include "net/strategies/FailoverStrategy.h"
 #include "net/strategies/SinglePoolStrategy.h"
 #include "net/Url.h"
@@ -46,7 +45,6 @@
 
 
 NonceMapper::NonceMapper(size_t id, xmrig::Controller *controller) :
-    m_donate(nullptr),
     m_suspended(0),
     m_pending(nullptr),
     m_id(id),
@@ -54,10 +52,6 @@
 {
     m_storage  = new NonceStorage();
     m_strategy = createStrategy(controller->config()->pools());
-
-    if (controller->config()->donateLevel() > 0) {
-        m_donate = new DonateStrategy(id, controller, this);
-    }
 }
 
 
@@ -66,7 +60,6 @@
     delete m_pending;
     delete m_strategy;
     delete m_storage;
-    delete m_donate;
 }
 
 
@@ -140,7 +133,7 @@
     JobResult req = event->request;
     req.diff = m_storage->job().diff();
 
-    IStrategy *strategy = m_donate && m_donate->isActive() ? m_donate : m_strategy;
+    IStrategy *strategy = m_strategy;
 
     m_results[strategy->submit(req)] = SubmitCtx(req.id, event->miner()->id());
 }
@@ -150,9 +143,6 @@
 {
     m_strategy->tick(now);
 
-    if (m_donate) {
-        m_donate->tick(now);
-    }
 }
 
 
@@ -197,10 +187,6 @@
                  m_id, client->host(), client->port(), job.diff());
     }
 
-    if (m_donate && m_donate->isActive() && client->id() != -1 && !m_donate->reschedule()) {
-        return;
-    }
-
     m_storage->setJob(job);
 }
 
@@ -219,7 +205,7 @@
 {
     const SubmitCtx ctx = submitCtx(result.seq);
 
-    AcceptEvent::start(m_id, ctx.miner, result, client->id() == -1, error);
+    AcceptEvent::start(m_id, ctx.miner, result, error);
 
     if (!ctx.miner) {
         return;
@@ -272,10 +258,6 @@
 {
     m_suspended = 0;
     m_strategy->connect();
-
-    if (m_donate) {
-        m_donate->connect();
-    }
 }
 
 
@@ -285,8 +267,4 @@
     m_storage->setActive(false);
     m_storage->reset();
     m_strategy->stop();
-
-    if (m_donate) {
-        m_donate->stop();
-    }
 }
diff -ruN xmrig-proxy/src/proxy/splitters/nicehash/NonceMapper.h xmrig-proxy_new/src/proxy/splitters/nicehash/NonceMapper.h
--- xmrig-proxy/src/proxy/splitters/nicehash/NonceMapper.h	2018-03-25 20:43:48.989371093 +0800
+++ xmrig-proxy_new/src/proxy/splitters/nicehash/NonceMapper.h	2018-03-25 21:12:45.467771324 +0800
@@ -34,7 +34,6 @@
 #include "net/Job.h"
 
 
-class DonateStrategy;
 class IStrategy;
 class JobResult;
 class LoginRequest;
@@ -97,7 +96,6 @@
     void connect();
     void suspend();
 
-    DonateStrategy *m_donate;
     int m_suspended;
     IStrategy *m_pending;
     IStrategy *m_strategy;
diff -ruN xmrig-proxy/src/proxy/splitters/simple/SimpleMapper.cpp xmrig-proxy_new/src/proxy/splitters/simple/SimpleMapper.cpp
--- xmrig-proxy/src/proxy/splitters/simple/SimpleMapper.cpp	2018-03-25 20:43:48.990371130 +0800
+++ xmrig-proxy_new/src/proxy/splitters/simple/SimpleMapper.cpp	2018-03-25 21:18:57.103774496 +0800
@@ -31,7 +31,6 @@
 #include "core/Controller.h"
 #include "log/Log.h"
 #include "net/Client.h"
-#include "net/strategies/DonateStrategy.h"
 #include "net/strategies/FailoverStrategy.h"
 #include "net/strategies/SinglePoolStrategy.h"
 #include "net/Url.h"
@@ -46,7 +45,6 @@
 
 SimpleMapper::SimpleMapper(uint64_t id, xmrig::Controller *controller) :
     m_active(false),
-    m_donate(nullptr),
     m_pending(nullptr),
     m_miner(nullptr),
     m_id(id),
@@ -65,7 +63,6 @@
 {
     delete m_pending;
     delete m_strategy;
-    delete m_donate;
 }
 
 
@@ -115,7 +112,7 @@
     JobResult req = event->request;
     req.diff = m_job.diff();
 
-    IStrategy *strategy = m_donate && m_donate->isActive() ? m_donate : m_strategy;
+    IStrategy *strategy = m_strategy;
 
     if (strategy) {
         strategy->submit(req);
@@ -131,9 +128,6 @@
         m_idleTime++;
     }
 
-    if (m_donate) {
-        m_donate->tick(now);
-    }
 }
 
 
@@ -166,10 +160,6 @@
                  m_id, client->host(), client->port(), job.diff());
     }
 
-    if (m_donate && m_donate->isActive() && client->id() != -1 && !m_donate->reschedule()) {
-        return;
-    }
-
     setJob(job);
 }
 
@@ -234,9 +224,6 @@
 {
     m_strategy->connect();
 
-    if (m_donate) {
-        m_donate->connect();
-    }
 }
 
 
diff -ruN xmrig-proxy/src/proxy/splitters/simple/SimpleMapper.h xmrig-proxy_new/src/proxy/splitters/simple/SimpleMapper.h
--- xmrig-proxy/src/proxy/splitters/simple/SimpleMapper.h	2018-03-25 20:43:48.990371130 +0800
+++ xmrig-proxy_new/src/proxy/splitters/simple/SimpleMapper.h	2018-03-25 21:17:03.241484261 +0800
@@ -34,7 +34,6 @@
 #include "net/Job.h"
 
 
-class DonateStrategy;
 class IStrategy;
 class JobResult;
 class LoginRequest;
@@ -83,7 +82,6 @@
 
     bool m_active;
     bool m_dirty;
-    DonateStrategy *m_donate;
     IStrategy *m_pending;
     IStrategy *m_strategy;
     Job m_job;
diff -ruN xmrig-proxy/src/proxy/Stats.cpp xmrig-proxy_new/src/proxy/Stats.cpp
--- xmrig-proxy/src/proxy/Stats.cpp	2018-03-25 20:43:48.989371093 +0800
+++ xmrig-proxy_new/src/proxy/Stats.cpp	2018-03-25 21:13:19.264044788 +0800
@@ -113,10 +113,6 @@
     m_data.accepted++;
     m_data.hashes += event->result.diff;
 
-    if (event->isDonate()) {
-        m_data.donateHashes += event->result.diff;
-    }
-
     Counters::accepted++;
 
     const size_t ln = m_data.topDiff.size() - 1;
@@ -131,9 +127,5 @@
 
 void Stats::reject(const AcceptEvent *event)
 {
-    if (event->isDonate()) {
-        return;
-    }
-
     m_data.rejected++;
 }
diff -ruN xmrig-proxy/src/proxy/StatsData.h xmrig-proxy_new/src/proxy/StatsData.h
--- xmrig-proxy/src/proxy/StatsData.h	2018-03-25 20:43:48.989371093 +0800
+++ xmrig-proxy_new/src/proxy/StatsData.h	2018-03-25 21:13:43.972975833 +0800
@@ -40,7 +40,6 @@
     inline StatsData() :
         accepted(0),
         connections(0),
-        donateHashes(0),
         expired(0),
         hashes(0),
         invalid(0),
@@ -91,7 +90,6 @@
     std::vector<uint16_t> latency;
     uint64_t accepted;
     uint64_t connections;
-    uint64_t donateHashes;
     uint64_t expired;
     uint64_t hashes;
     uint64_t invalid;
